circe/src/main/scala/org/http4s/circe/CirceSensitiveDataEntityDecoder.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2015 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.circe
+
+import cats.effect.Sync
+import io.circe.Decoder
+import org.http4s.EntityDecoder
+
+/** Derive [[EntityDecoder]] if implicit [[Decoder]] is in the scope without need to explicitly call `jsonOfSensitive`
+  *
+  * Note that it varies from [[CirceEntityDecoder]] in that, when failing to decode [[io.circe.Json]] to an `A`,
+  * the JSON will not be included in the raised [[Exception]]. In the event the JSON includes sensitive data, this trait is,
+  * arguably, a better choice since it eliminates the risk of logging sensitive data, e.g. due to logging a raised [[Throwable]]
+  * that includes the sensitive JSON.
+  */
+trait CirceSensitiveDataEntityDecoder {
```

Comment:
I'm not convinced I like this, when it's parameterized everywhere else.  But I also don't like the implicit JSON codecs anyway, so I might be overly grumpy.  I'm curious what other people think.

Reply:
> when it's parameterized everywhere else

Could you please say more, Ross? I don't follow.

Reply:
We've surfaced that `Json => String` everywhere else, but this one is hardcoded to the constant function.  And for it to be implicit, it must be hardcoded to some function.  You can't fix it and still retain the utility of this class.

I think implicit entity codecs from implicit circe codecs are a mistake, so I think my complaint is just a pre-existing condition here. :smile:

Reply:
> I think implicit entity codecs from implicit circe codecs are a mistake, so I think my complaint is just a pre-existing condition here. 😄

Gotcha, 😀! So, philosophically, you believe the codecs should be defined in the actual application￼.